WooCommerce/Classes/ViewRelated/Reviews/ReviewsView.swift

```diff
@@ -18,10 +19,18 @@ struct ReviewsView: UIViewControllerRepresentable {
     ///
     func makeUIViewController(context: Self.Context) -> ReviewsViewController {
         let viewController = ReviewsViewController(siteID: siteID)
+        // This makes sure that the navigation item of the hosting controller
+        // is in sync with that of the wrapped controller.
         context.coordinator.parentObserver = viewController.observe(\.parent, changeHandler: { vc, _ in
             vc.parent?.navigationItem.title = vc.title
             vc.parent?.navigationItem.rightBarButtonItems = vc.navigationItem.rightBarButtonItems
         })
+
+        // This fixes the issue when `rightBarButtonItem` is updated in `ReviewsViewController`,
+        // the hosting controller should be updated to reflect the change.
+        context.coordinator.rightBarButtonItemObserver = viewController.observe(\.navigationItem.rightBarButtonItem, changeHandler: { vc, _ in
+            vc.parent?.navigationItem.rightBarButtonItems = vc.navigationItem.rightBarButtonItems
```

Comment:
I was wondering why it updates the parent's `navigationItem.rightBarButtonItems` (plural) instead of `navigationItem.rightBarButtonItem` (singular) since it's observing `viewController.rightBarButtonItem`? It seems like updating `rightBarButtonItems` is good too (probably the same internally when there is only one item), just curious 😄

Reply:
Good catch! Actually, I think it makes more sense to just update `rightBarButtonItem`, although `rightBarButtonItems` works as well. Updated in 4b43c3d599bec5bc294441660129236f0ba3354e 👍